GDJS/Runtime/runtimeobject.ts

```diff
@@ -1344,7 +1346,20 @@ namespace gdjs {
      * @param multiplier Set the force multiplier
      */
     addForce(x: float, y: float, multiplier: integer): void {
-      this._forces.push(this._getRecycledForce(x, y, multiplier));
+      if (multiplier === 1) {
+        this._permanentForce.add(x, y);
+      } else if (
+        multiplier === 0 &&
+        this._instantForces.length > 0 &&
+        this._instantForces[0].getMultiplier() === 0
+      ) {
+        // Avoid to instantiate new a Force for each instance force.
+        this._instantForces[0].add(x, y);
```

Comment:
Nice. You could almost have had a `_permanentForce`, a `_instantForce` and a `_legacyForces`.

Reply:
Yes, it would be clearer, but it would take a bit more memory for objects with no force.

Reply:
We could maybe to something like a "ForceManager" with a lazy instantiation to reduce memory usage even more, but we can come back to this later.

Reply:
That's very true. Let's see later and it's true that avoiding more memory for each instance is a good thing to ensure.